third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -151,6 +152,22 @@ func resourceComputeSecurityPolicy() *schema.Resource {
 	}
 }
 
+func rulesCustomizeDiff(diff *schema.ResourceDiff, _ interface{}) error {
+	_, n := diff.GetChange("rule")
+	nSet := n.(*schema.Set)
+
+	nPriorities := map[int64]bool{}
+	for _, rule := range nSet.List() {
+		priority := int64(rule.(map[string]interface{})["priority"].(int))
```

Comment:
This will panic if the values aren't set correctly - is that okay?  I'm not sure what invariants TF provides for types, unfortunately.

Reply:
Terraform should verify that `priority` is an `int` before we get here.  Is that what you're asking?  Very similar code is used in the update function and it seems we haven't hit any problems with that yet (https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_compute_security_policy.go#L254), however if you feel more comfortable with double-checking it, I can certainly throw some checks around here.